internal/backend/b2/b2.go

```diff
@@ -258,7 +261,23 @@ func (be *b2Backend) Remove(ctx context.Context, h restic.Handle) error {
 	defer be.sem.ReleaseToken()
 
 	obj := be.bucket.Object(be.Filename(h))
-	return errors.Wrap(obj.Delete(ctx), "Delete")
+
+	if be.canDelete {
+		delete_err := obj.Delete(ctx)
+		err := errors.Wrap(delete_err, "Delete")
+		if err != nil {
```

Comment:
That check is still broken (iff delete_err != nil, then err != nil). Just remove this `if` then it should work.

Reply:
Ugh indeed that's all wrong still.

I don't think just removing the `if` would work. I'm not sure but`base.Code(nil)` probably wouldn't work. So instead I just return early in the nil case now.

I've also added a log for the fallback case so this should be easier to observe working now. Though again I haven't gotten around to testing yet.